src/js/angular/rest/graphql.rest.service.js

```diff
@@ -175,6 +175,25 @@ function GraphqlRestService($http) {
         return $http.get(`${REPOSITORIES_ENDPOINT}/${repositoryId}/graphql/manage/endpoints/${endpointId}/export`, {responseType: 'blob'});
     }
 
+    /**
+     * Import the endpoint definitions, either as a single file or as multiple files where the file type can be either
+     * yaml or zip.
+     * @param {string} repositoryId The repository ID.
+     * @param {FormData|File} payload The request payload.
+     * @param {*} canceler The canceler object.
+     * @returns {Promise<unknown>} The response from the backend.
+     */
+    const importEndpointDefinition = (repositoryId, payload, canceler) => {
+        return Upload.http({
+            url: `${REPOSITORIES_ENDPOINT}/${repositoryId}/graphql/manage/endpoints/import`,
+            data: payload,
+            headers: {
+                'Content-Type': undefined
```

Comment:
Why is it undefined?

Reply:
This allows the type to be obtained by the browser. Not sure if it's strictly mandatory though.

Reply:
Yeah, the content-type header must be set, and as far as we allow different file types to be imported, it's easier to set it this way and allow the appropriate type to be set from the browser.